include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_merge_sort.h

```diff
@@ -63,7 +63,7 @@ struct __group_merge_path_sorter
     template <typename _StorageAcc, typename _Compare>
     bool
     sort(const sycl::nd_item<1>& __item, const _StorageAcc& __storage_acc, _Compare __comp, std::uint32_t __start,
-         std::uint32_t __end, std::uint32_t __sorted, std::uint16_t __data_per_workitem,
+         std::uint32_t __end, std::uint32_t __sorted, std::uint32_t __data_per_workitem,
```

Comment:
Last minor question:  
What changed here to need this type change?  Is it a bug fix?  it seems unrelated to the rest of the PR.

Reply:
After offline discussion, we see that `__data_per_workitem` is only used in 32 bit unsigned context, and we have made things more explicit by removing `auto` in the return from `__find_start_point` for the other parameters with `_Index` type going into `__serial_merge` so that it is clear that we will end up with a consistent type.  
Thanks for the changes here.